store/store.go

```diff
@@ -956,11 +922,10 @@ type SnapSpec struct {
 
 // SnapInfo returns the snap.Info for the store-hosted snap matching the given spec, or an error.
 func (s *Store) SnapInfo(snapSpec SnapSpec, user *auth.UserState) (*snap.Info, error) {
-	// get the query before doing Parse, as that overwrites it
-	query := s.detailsURI.Query()
-	u, err := s.detailsURI.Parse(snapSpec.Name)
-	if err != nil {
-		return nil, err
+	query := url.Values{}
+
+	if len(s.detailFields) != 0 {
+		query.Set("fields", encodeFields(s.detailFields))
 	}
```

Comment:
wondering if instead of the global helper encodeFields, this bit shouldn't be some kind of helper on Store itself

Reply:
I extracted this setup to Store.defaultSnapQuery, thanks.